systems/notification/event-notify/pkg/server/event.go

```diff
@@ -52,684 +52,743 @@ func (es *EventToNotifyEventServer) EventNotification(ctx context.Context, e *ep
-				c := evt.EventToEventConfig[evt.EventHealthCappStore]
-				// msg, err := epb.Unmarshal(e.Msg, c.Name)
-				// if err != nil {
-				// 	return nil, err
-				// }
-			case msgbus.PrepareRoute(es.orgName, evt.EventRoutingKey[evt.EventNotificationStore]):
-				//c := evt.EventToEventConfig[evt.EventNotificationStore]
-				// msg, err := epb.Unmarshal(e.Msg, c.Name)
-				// if err != nil {
-				// 	return nil, err
-				// }
-		*/
-	default:
-		log.Errorf("No handler routing key %s", e.RoutingKey)
+	dynamicConfig.Type = notificationType
+
+	es.processEvent(&dynamicConfig, es.orgId, "", msg.NodeId, "", "", jmsg, msg.NodeId)
+}
+
+func handleEventPaymentSuccess(es *EventToNotifyEventServer, msg *epb.Payment, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
 	}
 
-	return &epb.EventResponse{}, nil
+	if msg.ItemType != ukama.ItemTypeInvoice.String() {
+		log.Errorf("unexpected item type for successful payment: %s", msg.ItemType)
+		return
+	}
+
+	metadata := map[string]string{}
+
+	err = json.Unmarshal(msg.Metadata, &metadata)
+	if err != nil {
+		log.Errorf("failed to Unmarshal payment metadata as map[string]string: %v", err)
+	}
+
+	targetId, ok := metadata["targetId"]
+	if !ok {
+		log.Errorf("missing targetId metadata for successful package payment: %s", msg.ItemId)
+		return
+	}
+
+	es.processEvent(c, es.orgId, "", "", targetId, targetId, jmsg, msg.Id)
+}
+
+func handleEventPaymentFailed(es *EventToNotifyEventServer, msg *epb.Payment, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
+
+	if msg.ItemType != ukama.ItemTypePackage.String() {
+		log.Errorf("unexpected item type for successful payment: %s", msg.ItemType)
+		return
+	}
+
+	metadata := map[string]string{}
+
+	err = json.Unmarshal(msg.Metadata, &metadata)
+	if err != nil {
+		log.Errorf("failed to Unmarshal payment metadata as map[string]string: %v", err)
+	}
+
+	targetId, ok := metadata["targetId"]
+	if !ok {
+		log.Errorf("missing targetId metadata for successful package payment: %s", msg.ItemId)
+		return
+	}
+
+	es.processEvent(c, es.orgId, "", "", targetId, targetId, jmsg, msg.Id)
+}
+
+func handleEventInvoiceGenerate(es *EventToNotifyEventServer, msg *epb.Report, c *evt.EventConfig) {
+	jmsg, err := json.Marshal(msg)
+	if err != nil {
+		log.Errorf("Failed to store raw message for %s to db. Error %+v", c.Name, err)
+	}
+	es.processEvent(c, es.orgId, msg.NetworkId, "", "", "", jmsg, msg.Id)
 }
 
-func (es *EventToNotifyEventServer) ProcessEvent(ec *evt.EventConfig, orgId, networkId, nodeId, subscriberId, userId string, msg []byte, rid string) *db.Notification {
+func (es *EventToNotifyEventServer) processEvent(ec *evt.EventConfig, orgId, networkId, nodeId, subscriberId, userId string, msg []byte, rid string) *db.Notification {
```

Comment:
Also, Why are we only logging errors in `processEvent`  and all the `handleEventXXXX`  functions? shouldn't we communicate these errors back? I am not sure if this choice is somewhat related to notification handling, so let me know.